### ISSUE TITLE
code1 

### ISSUE THREAD

**Issue by Ruthwik333** [HarshitKmr10/somethingsomething#1]
**important**
For each function comment the function what is the function intend to do
comments hyper parameter tuning

Predicted Topic : Nan
**important**
Threshold true or false
Threshold 
import pandas as pd
import numpy as np
import string
import re
import unicodedata
from sklearn.preprocessing import LabelEncoder
from sklearn.pipeline import Pipeline
from sklearn.feature_extraction.text import TfidfVectorizer
import lightgbm as lgb
from sklearn.metrics import accuracy_score as acc
import joblib

LE = LabelEncoder()
df = pd.read_excel("traindf.xlsx")
df['Meeting Notes'] = df['Meeting Notes'].astype(str)
unique_topics = df['Topic'].unique()
print(unique_topics)

def encoding(text):
    if isinstance(text, str):
        text = unicodedata.normalize("NFKD", text)
    return text

def remove_URL(text):
    return re.sub(r"https?://\S+|www.\S+", "", text)

def remove_non_ascii(text):
    return re.sub(r'[^\x00-\x7f]', r'', text)

def remove_html(text):
    html = re.compile(r"<.*?>|&([a-z0-9]+|#[0-9]{1,6}|#x[0-9a-f]{1,6});")
    return re.sub(html, "", text)

def remove_punct(text):
    return text.translate(str.maketrans('', '', string.punctuation))

print(df.shape)
print(df.head)
df = df[['Meeting Notes', "Topic"]]

df.dropna(inplace=True)  # Added inplace=True to modify the DataFrame
print(df.shape)
df['Clean'] = df['Meeting Notes'].apply(remove_URL)
df['Clean'] = df['Clean'].apply(remove_non_ascii)
df['Clean'] = df['Clean'].apply(remove_html)

le = LE.fit(df["Topic"])
df["Topic"] = le.transform(df["Topic"])
x = df["Clean"]
y = df["Topic"]

lgb_model = Pipeline((("vect", TfidfVectorizer()), ("clc", lgb.LGBMClassifier(boosting_type='gbdt', objective='multiclass'))))
lgb_model.fit(x, y)

threshold = 0.90

train_probabilities = lgb_model.predict_proba(x)
topic_names = le.inverse_transform(np.arange(len(lgb_model.classes_)))
result_df = pd.DataFrame(train_probabilities, columns=topic_names)

predicted_classes = [topic_names[i] for i in np.argmax(train_probabilities, axis=1)]
filtered_classes = [cls if prob > threshold else 'Other' for cls, prob in zip(predicted_classes, np.max(train_probabilities, axis=1))]

df["predictedtopic"] = filtered_classes
df = pd.concat([df, result_df], axis=1)
df = df[df["predictedtopic"] != "Other"]
train_acc = acc(le.transform(df["predictedtopic"]), df['Topic'])

# Inverse transform the 'Topic' column back to its original values
df['Topic'] = le.inverse_transform(df['Topic'])
#df['predictedtopic'] = le.inverse_transform(le.transform(df['predictedtopic']))

# Rename the probability columns with topic names
for topic in topic_names:
    df[f'Probability_{topic}'] = df[topic]

# Export the DataFrame to an Excel file
df.drop(columns=topic_names, inplace=True)  # Remove redundant columns
df.to_excel("MFAtrainingfinal.xlsx", index=False)
print("Train Accuracy:", train_acc)
print(topic_names)
# Saving the model
joblib.dump(lgb_model, "model.pkl")
